src/file/file.rpc.ts

```diff
@@ -26,52 +26,57 @@ const packageDefinition = protoLoader.loadSync(
     oneofs: true,
   });
 
+// Has the full package hierarchy
+const protoDescriptor = grpc.loadPackageDefinition(packageDefinition);
+const file_proto : any = protoDescriptor.file;
+
 export const serviceNames: string[] = ['', 'file.fileService'];
 export const healthCheckStatusMap = {
   '': HealthCheckResponse.ServingStatus.UNKNOWN,
   serviceName: HealthCheckResponse.ServingStatus.UNKNOWN
 };
 
-// Has the full package hierarchy
-const protoDescriptor = grpc.loadPackageDefinition(packageDefinition);
-const file_proto = protoDescriptor.file;
-
 /**
  * The FileServer class, containing all of the FileServer methods.
  */
 export class FileServer {
   public server: any;
   public grpcHealthCheck: GrpcHealthCheck;
+
   public constructor(port: string) {
     this.server = new grpc.Server();
+    this.addServices();
+    this.server.bind(`0.0.0.0:${port}`, grpc.ServerCredentials.createInsecure());
+  }
 
+  private addServices () {
     // Register the health service
     this.grpcHealthCheck = new GrpcHealthCheck(healthCheckStatusMap);
     this.server.addService(HealthService, this.grpcHealthCheck);
-    this.server.addService(
-      file_proto.FileService.service,
-      {
-        GenerateKey: this.wrapper(this.generateKey),
-        CreateUpload: this.wrapper(this.createUpload),
-        UpdateUploadID: this.wrapper(this.updateUpload),
-        GetUploadByID: this.wrapper(this.getUploadByID),
-        DeleteUploadByID: this.wrapper(this.deleteUploadByID),
-        GetFileByID: this.wrapper(this.getFileByID),
-        GetFileByKey: this.wrapper(this.getFileByKey),
-        GetFilesByFolder: this.wrapper(this.getFilesByFolder),
-        CreateFile: this.wrapper(this.createFile),
-        DeleteFile: this.wrapper(this.deleteFile),
-        IsAllowed: this.wrapper(this.isAllowed),
-      }
-    );
-    this.server.bind(`0.0.0.0:${port}`, grpc.ServerCredentials.createInsecure());
+
+    const services = {
+      GenerateKey: this.wrapper(this.generateKey),
+      CreateUpload: this.wrapper(this.createUpload),
+      UpdateUploadID: this.wrapper(this.updateUpload),
+      GetUploadByID: this.wrapper(this.getUploadByID),
+      DeleteUploadByID: this.wrapper(this.deleteUploadByID),
+      GetFileByID: this.wrapper(this.getFileByID),
+      GetFileByKey: this.wrapper(this.getFileByKey),
+      GetFilesByFolder: this.wrapper(this.getFilesByFolder),
+      CreateFile: this.wrapper(this.createFile),
+      DeleteFile: this.wrapper(this.deleteFile),
+      IsAllowed: this.wrapper(this.isAllowed),
+    };
+
+    this.server.addService(file_proto.FileService.service, services);
   }
 
-  private wrapper (func: any) : any {
+  private wrapper (rpcFunction: any) : any {
```

Comment:
The functions that are passed to `wrapper` are not "rpc" functions, they are just service functions, this name still doesn't fit.

Reply:
so how would you call it?

Reply:
Call it `func` and give it a type of `Function`.

Reply:
Done.

Reply:
@Shahar-Y Just `func`, and give it a type of `Function`.